react-components/src/util/dialog/DEConfirmationDialog.js

```diff
@@ -27,33 +27,33 @@ class DEConfirmationDialog extends Component {
             okLabel,
             messages
         } = this.props;
+        const dialogTitleID = build(debugId, "title");
+
         return (
-            <Dialog open={dialogOpen} id={debugId}>
-                <DEDialogHeader
-                    messages={messages}
-                    heading={heading}
-                    onClose={() => {
-                        if (onCancelBtnClick) {
-                            onCancelBtnClick();
-                        }
-                    }}/>
+            <Dialog id={debugId}
+                    open={dialogOpen}
+                    aria-labelledby={dialogTitleID}
+            >
+                <DEDialogHeader id={dialogTitleID}
+                                messages={messages}
+                                heading={heading}
+                                onClose={onCancelBtnClick}
+                />
```

Comment:
`onCancelBtnClick` is not a required prop. Should we add back the check to see if the method is `undefined`? Actually let's make it required.

Reply:
We don't have to check for `null`/`undefined`, since this is being passed directly as the `onClick` prop to `IconButton`, and it handles `null`/`undefined` for its `onClick` just fine.

Since `onClose` is a required prop for `DEDialogHeader`, then I agree that it should be required for `DEConfirmationDialog` as well, so I'll update that in this PR.

Let's also consider updating `DEConfirmationDialog`, in a future PR, to make the `OK` or `Cancel` button optional, so we can show a dialog with only 1 "confirm" or "cancel" button (we may have already discussed this before in elsewhere 🤔 ).

Reply:
Weird...I do get 
`analysis/view/with test analyses/error: ["Uncaught TypeError: onCancelBtnClick is not a function", "http://localhost:6006/vendors~main.369c8f96462db18bc2f3.bundle.js", 107711, 5, TypeError]
analysis/view/with test analyses/error: ["Uncaught TypeError: onCancelBtnClick is not a function", "http://localhost:6006/main.369c8f96462db18bc2f3.hot-update.js", 116, 11, TypeError]` when I don't pass the prop in SB. 🤔 

Anyway I agree this should be a required prop. 

WRT `DEConfirmationDialog` we agreed to make `OK` and `Cancel` text as prop. But yes we can make just one button mandatory allowing us to display the dialog with 1 button.